public/app/plugins/datasource/elasticsearch/query_builder.ts

```diff
@@ -367,4 +367,31 @@ export class ElasticQueryBuilder {
 
     return query;
   }
+
+  getLogsQuery(target, querystring) {
```

Comment:
Would be nice with a test of this verifying that both documents and date histogram is requested.